### ISSUE TITLE
VS incorrectly report a build as succeeded when a Roslyn code generator warning CS8785 treated as error fails the build

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#65313]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/VS-incorrectly-report-a-build-as-succeed/10096824)._

---
I am using VS 17.2.6 and also a recent Int Preview.
I have a class library project that uses a Roslyn code generator (from a nuget package) and targets net472.
I have `$(WarningsAsErrors);CS8785` in my csproj file.
When I build the project I correctly get this in the output log:
```
1>CSC : error CS8785: Generator 'SdkJsonSourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.Collections.Immutable.ImmutableArray`1 Microsoft.CodeAnalysis.GeneratorExecutionContext.get_AdditionalFiles()'.'
```
But the project is incorrectly marked as succeeded at the end of the output log:
```
1>CommentRemover -> C:\repos\GitHubVSExtensibility\bin\samples\CommentRemover\Debug\net472\CommentRemover.dll
1>Done building project "CommentRemover.csproj".
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```
The error shows up correctly in the "Error List"
![image](https://user-images.githubusercontent.com/12466233/200926626-dec2d25d-8f29-4d7a-acac-b8baea4c9ce7.png)

If I build the project from command line using `msbuild` the build is correctly reported as failed:
```
"C:\repos\GitHubVSExtensibility\New_Extensibility_Model\Samples\CommentRemover\CommentRemover.sln" (default target) (1) ->
"C:\repos\GitHubVSExtensibility\New_Extensibility_Model\Samples\CommentRemover\CommentRemoverContainer\CommentRemoverContainer.csproj" (default target) (2) ->
"C:\repos\GitHubVSExtensibility\New_Extensibility_Model\Samples\CommentRemover\CommentRemover\CommentRemover.csproj" (default target) (3:3) ->
(CoreCompile target) ->
  CSC : error CS8785: Generator 'SdkJsonSourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.Collections.Immutable.ImmutableArray`1 Microsoft.CodeAnalysis.GeneratorExecutionContext.get_AdditionalFiles()'.' [C:\repos\GitHubVSExtensibility\New_Extensibility_Model\Samples\CommentRemover\CommentRemover\CommentRemover.csproj]

    132 Warning(s)
    1 Error(s)
```
